src/lib.rs

```diff
@@ -2362,6 +2370,11 @@ impl Connection {
 
     /// Selects the packet number space for outgoing packets.
     fn write_epoch(&self) -> Result<packet::Epoch> {
+        // On error during handshake, latest space is always Initial
+        if self.error.is_some() && !self.handshake_completed {
+            return Ok(packet::EPOCH_INITIAL);
```

Comment:
Why we want to go back to Initial? I think we simply need to follow current crypto level. If we are in Application or Handshake level already, going back to Initial doesn't guarantee the packet is decrypted properly. https://tools.ietf.org/html/draft-ietf-quic-tls-20#section-4.10